programs/manifest/src/program/processor/shared.rs

```diff
@@ -157,3 +166,48 @@ pub fn get_dynamic_value<T: Get>(data: &[u8]) -> DynamicAccount<T, Vec<u8>> {
     };
     dynamic_account
 }
+
+// TODO: Make dynamic_account immutable here
+pub(crate) fn get_trader_index_with_hint(
+    trader_index_hint: Option<DataIndex>,
+    dynamic_account: &mut MarketRefMut,
+    payer: &Signer,
+) -> Result<DataIndex, ProgramError> {
+    let trader_index: DataIndex = match trader_index_hint {
+        None => dynamic_account.get_trader_index(payer.key),
+        Some(hinted_index) => {
+            verify_trader_index_hint(hinted_index, &dynamic_account, &payer)?;
+            hinted_index
+        }
+    };
+    Ok(trader_index)
+}
+
+fn verify_trader_index_hint(
+    hinted_index: DataIndex,
+    dynamic_account: &MarketRefMut,
+    payer: &Signer,
+) -> ProgramResult {
+    require!(
+        hinted_index % (MARKET_BLOCK_SIZE as DataIndex) == 0,
+        ManifestError::WrongIndexHintParams,
+        "Invalid trader hint index {}",
```

Comment:
would favor more descriptive messages here:
- hint index not aligned
- expected claimed seat at hint index 
- expected claimed seat at hint index to be owned by payer

Reply:
Just the message, or need a different error too?